test/bc_coeff_compositions.jl

```diff
@@ -216,21 +220,21 @@ end
     ghost_u = A \ u
 
     # Check that A\u.(x) is consistent with analytic_AL \ u.(x)
-    @test analytic_u ≈ ghost_u

```

Comment:
that's not good.

Reply:
whoops! Don't know how this got deleted

Reply:
I haven't had the chance to play around with this branch, but my hunch is that some change to the convolutions is impacting at least one of `L*Q.L` or `L*Q.b`, where `Q.L` is the linear part of `Q` and `Q.b` is the affine part of `Q`.

Reply:
Playing around with the branch, it seems that the left division `analytic_AL \  (u - analytic_Ab)` versus `Array(A)[1] \ (u - Array(A)[2])` is the issue. I've swapped the `analytic_Ab` and `Array(A)[2]` in the above computations and the result is always equivalent to `analytic_u`, but I swap `analytic_A` and `Array(A)[1]` I then start obtain varying results. 

I do get that `analytic_AL ≈ Array(A)[1]`, however, they do differ slightly in a few indices, which may be propagating causing numerical errors to propagate through.

Reply:
@ChrisRackauckas @shivin9 It appears that the stencil fix is causing a discrepancy in `L`'s high boundary coefficients with those of the analytic stencil's high  boundary coefficients. There is also a discrepancy in one position of the low boundary coefficients, but this is not causing any differences to explode. 

The additional thing I noticed is that `L`'s high and low boundary coefficients are not exactly reflections of each other, there is a slight discrepancy in the 4th and 5th positions. Although the stencils are approximately equivalent. 

This may be caused due to a combination of:
- moving the handling of dx into the Fornberg algorithm, which may be causing the computations to accumulate errors. Perhaps, we should compute the stencils without dx being incorporated, and then dividing by dx^derivative_order? Which is a more stable implementation? 
- This example has dx = 0.01 and derivative order equal to 4. This will magnify small differences in stencils, though, this shouldn't be a problem.

Reply:
```
julia> L.low_boundary_coefs
L2-element StaticArrays.SArray{Tuple{2},StaticArrays.SArray{Tuple{8},Float64,1,8},1,2}:
. [3.5e8, -1.86667e9, 4.25e9, -5.4e9, 4.18333e9, -2.0e9, 5.5e8, -6.66667e7]        
 [6.66667e7, -1.83333e8, 0.0, 5.16667e8, -7.33333e8, 4.5e8, -1.33333e8, 1.66667e7]

julia> L.high_boundary_coefs
2-element StaticArrays.SArray{Tuple{2},StaticArrays.SArray{Tuple{8},Float64,1,8},1,2}:
 [1.66667e7, -1.33333e8, 4.5e8, -7.33333e8, 5.16667e8, 0.0, -1.83333e8, 6.66667e7]
 [-6.66667e7, 5.5e8, -2.0e9, 4.18333e9, -5.4e9, 4.25e9, -1.86667e9, 3.5e8]        

```

I'm getting exact mirror images. Can you post an MWE?

Plus the newer implementation is giving reduced errors so it has to be right.

Reply:
I obtain the following:

`julia> L = CenteredDifference(4, 4, 0.01, 20)` 

`julia> L.low_boundary_coefs[1] .- reverse(L.high_boundary_coefs[2])`
`8-element StaticArrays.SArray{Tuple{8},Float64,1,8}:`
`  0.0               `
`  0.0               `
`  0.0               `
`  9.5367431640625e-7`
` -9.5367431640625e-7`
`  0.0               `
`  0.0               `
`  0.0              `

Reply:
I don't disagree that the newer implementation is more accurate, I'm just a bit confused why our tests are failing (or are so sensitive to minute differences).